src/justfile.rs

```diff
@@ -311,43 +316,61 @@ impl<'src> Justfile<'src> {
     let mut evaluator =
       Evaluator::recipe_evaluator(context.config, dotenv, &scope, context.settings, search);
 
-    for Dependency { recipe, arguments } in recipe.dependencies.iter().take(recipe.priors) {
-      let arguments = arguments
-        .iter()
-        .map(|argument| evaluator.evaluate_expression(argument))
-        .collect::<RunResult<Vec<String>>>()?;
-
-      Self::run_recipe(
-        context,
-        recipe,
-        &arguments.iter().map(String::as_ref).collect::<Vec<&str>>(),
-        dotenv,
-        search,
-        ran,
-      )?;
-    }
+    let run_dependencies_in_parallel = recipe.attributes.contains(&Attribute::Parallel);
+    parallel::task_scope(run_dependencies_in_parallel, |scope| {
+      for Dependency { recipe, arguments } in recipe.dependencies.iter().take(recipe.priors) {
+        let arguments = arguments
+          .iter()
+          .map(|argument| evaluator.evaluate_expression(argument))
+          .collect::<RunResult<Vec<String>>>()?;
+
+        scope.run(move || {
+          Self::run_recipe(
+            context,
+            recipe,
+            &arguments.iter().map(String::as_ref).collect::<Vec<&str>>(),
+            dotenv,
+            search,
+            ran,
+          )
+        })?;
+      }
+      Ok(())
+    })?;
 
     recipe.run(context, dotenv, scope.child(), search, &positional)?;
 
     {
-      let mut ran = BTreeSet::new();
+      let ran = Ran::new();
 
-      for Dependency { recipe, arguments } in recipe.dependencies.iter().skip(recipe.priors) {
-        let mut evaluated = Vec::new();
+      parallel::task_scope(run_dependencies_in_parallel, |scope| {
+        for Dependency { recipe, arguments } in recipe.dependencies.iter().skip(recipe.priors) {
+          let mut evaluated = Vec::new();
 
-        for argument in arguments {
-          evaluated.push(evaluator.evaluate_expression(argument)?);
-        }
+          for argument in arguments {
+            evaluated.push(
+              evaluator
+                .evaluate_expression(argument)
+                .expect("error evaluating expression"),
```

Comment:
`task_scope` should accept a function that returns an error, so we can use `?` here.